client/verta/verta/pipeline/_pipeline_step.py

```diff
@@ -0,0 +1,285 @@
+        )
+
+    def set_registered_model_version(
+        self, registered_model_version: RegisteredModelVersion
+    ) -> RegisteredModelVersion:
+        """Set the registered model version associated with this step.
+
+        Parameters
+        ----------
+        registered_model_version : :class:`~verta.registry.entities.RegisteredModelVersion`
+            Registered model version to use for the step.
+
+        Returns
+        -------
+        RegisteredModelVersion
+            The new registered model version now set for this step.
+
+        Raises
+        ------
+        TypeError
+            If the provided value is not type RegisteredModelVersion.
+
+        """
+        if not isinstance(registered_model_version, RegisteredModelVersion):
+            raise TypeError(
+                f"registered_model_version must be a RegisteredModelVersion object, not {type(registered_model_version)}"
+            )
+        self._registered_model_version = registered_model_version
+        self._registered_model = self._get_registered_model()
+        return self.registered_model_version
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @name.setter
+    def name(self, value) -> None:
+        """Raise a more informative error than the default."""
+        raise AttributeError("can't set attribute 'name'; please use set_name()")
+
+    def set_name(self, name: str) -> str:
+        """Change the name of this step.
+
+        Parameters
+        ----------
+        name : str
+            New name to use for the step.
+
+        Returns
+        -------
+        str
+            The new name now set for this step.
+
+        Raises
+        ------
+        TypeError
+            If the provided value for ``name`` is not type str.
+
+        """
+        if not isinstance(name, str):
+            raise TypeError(f"name must be a string, not {type(name)}")
+        self._name = name
+        return self.name
+
+    @property
+    def predecessors(self) -> Set["PipelineStep"]:
+        return self._predecessors
+
+    @predecessors.setter
+    def predecessors(self, value) -> None:
+        """Raise a more informative error than the default."""
+        raise AttributeError(
+            "can't set attribute 'predecessors'; please use set_predecessors()"
+        )
+
+    def set_predecessors(
+        self,
+        steps: Optional[
+            Union[List["PipelineStep"], Set["PipelineStep"], Tuple["PipelineStep"]]
+        ] = None,
+    ) -> Set["PipelineStep"]:
+        """Set the predecessors associated with this step.
+
+        Parameters
+        ----------
+        steps : list, set, or tuple, optional
+            List, set, or tuple of PipelineStep objects whose outputs will be treated as
+            inputs to this step. All options are converted to a set, so order is irrelevant
+            and duplicates are removed. An empty set used if no input is provided.
```

Comment:
```suggestion
            and duplicates are removed. An empty set is used if no input is provided.
```

Reply:
Arg!  I know I made that change.  It must have gotten lost in a revert at some point.  My bad.  Fixing now